lib/logreader.c

```diff
@@ -449,11 +449,12 @@ log_reader_handle_line(LogReader *self, const guchar *line, gint length, LogTran
 {
   LogMessage *m;
 
+  m = log_msg_new_empty();
   msg_debug("Incoming log entry",
-            evt_tag_printf("line", "%.*s", length, line));
-  /* use the current time to get the time zone offset */
-  m = log_msg_new((gchar *) line, length,
-                  &self->options->parse_options);
+            evt_tag_printf("line", "%.*s", length, line),
+            evt_tag_msg_reference(m));
+
+  msg_format_parse(&self->options->parse_options, m, line, length);
```

Comment:
Using `log_msg_new_empty()` + `msg_format_parse()` instead of the original `log_msg_new()` might have some minor performance related side effects (for example, it allocates the payload a bit differently).

Is this change intentional? I've found it in a commit that is meant to introduce rcptid reporting only (`logreader: report rcptid in the "Incoming log entry" message`).

Reply:
The change is needed, as it's difficult to report msg related fields in a log message if the message does not exist yet :)

But good point on the allocation difference, I didn't notice that (I should have though). I am fixing that in a moment.

Reply:
I have just pushed another round that fixes the allocation issue you pointed out rightly.

It also fixes a circular dependency between msg-format.{c,h} and logmsg.{c,h} by moving the log_msg_new() functionality to become a method of MsgFormat. It also decreases test program dependency on log message parsing (and thus the syslogformat module) by using lower-level LogMessage functions. The latter is not complete but does away with a few easy-to-get-rid of cases.

Reply:
Thank you. 👍🏻